net/gajim/Portfile

```diff
@@ -16,37 +17,73 @@ long_description    The goal of Gajim is to provide a full featured and \
 
-master_sites        http://gajim.org/downloads/${branch}/
-use_bzip2           yes
+checksums           rmd160  c3573a515ceb3d53369b4c08a734d5785c77dd0e \
+                    sha256  5141ea1c27f4d2c1936de136dc29289a5ebf41e567dda8321e29d96518717203 \
+                    size    9578053
 
-checksums           rmd160  4bab0b953a6f9a362b43232f584a2d9bf372ef59 \
-                    sha256  46d4887857455b6e104d0478735edb51569bfa384578ffb4f4da81f1649b6a32
+depends_build       port:python38 \
+                    port:py38-setuptools \
+                    port:gettext
 
-depends_build       port:intltool \
-                    port:pkgconfig \
-                    port:autoconf \
-                    port:automake \
-                    port:libtool
+depends_run         port:py38-packaging \
+                    port:py38-nbxmpp \
+                    port:py38-openssl \
+                    port:py38-keyring \
+                    port:py38-cairo \
+                    port:py38-tinycss 
 
-depends_lib         path:lib/pkgconfig/gtk+-2.0.pc:gtk2 \
-                    port:gtkspell2 \
-                    port:py27-pygtk \
-                    port:py27-docutils \
-                    port:py27-nbxmpp \
-                    port:py27-openssl \
-                    port:dbus-python27
+depends_lib         path:lib/pkgconfig/gtk+-3.0.pc:gtk3 \
+                    port:glib2 \
+                    port:libsoup \
+                    port:py38-gobject3
 
-set python_prefix   ${frameworks_dir}/Python.framework/Versions/2.7
-configure.python    ${prefix}/bin/python2.7
-configure.pkg_config_path  ${python_prefix}/lib/pkgconfig
+post-patch {
+    reinplace "s|/usr/bin/env python3|/opt/local/bin/python3.8|" \
+        ${worksrcpath}/setup.py
+}
 
-# reconfigure using autogen.sh for intltool 0.51 compatibility
+set python.version  38
+set python.branch   [string range ${python.version} 0 end-1].[string index ${python.version} end]
+set python.prefix   ${frameworks_dir}/Python.framework/Versions/${python.branch}
+configure.python    ${prefix}/bin/python3.8
+configure.pkg_config_path  ${python.prefix}/lib/pkgconfig
 
-configure.cmd       ./autogen.sh
+variant audiovideo {
+    depends_run port:gstreamer1 \
+                port:gstreamer1-gst-libav \
+                port:gstreamer1-gst-plugins-base \
+                port:gstreamer1-gst-plugins-good \
+                port:gstreamer1-gst-plugins-ugly \
+                port:farstream
+    }
 
-livecheck.type      regex
-livecheck.url       ${homepage}/downloads.php?lang=en
-livecheck.regex     "Latest version of Gajim is\\s+<strong>(\\d+.\\d+.\\d+)<"
+variant geo {
+    depends_run port:geoclue2
+    }
+
+variant avatars {
+    depends_run port:py38-Pillow
+    }
+
+variant remote {
+    depends_run port:dbus
+    }
+
+variant nat {
+    depends_run port:gupnp-igd
+    }
+
+variant zeroconf {
+    depends_run port:avahi \
+                port:py27-pybonjour
+    # there is no py38--pybonjour atm
```

Comment:
this should likely not be the PY27 version?

Reply:
@reneeotten We only got this one: https://ports.macports.org/port/py-pybonjour/details/

Reply:
then you either remove the variant or add the missing package. This will never work since you use Python 3.8 and them add a PY27 package here...

Reply:
> then you either remove the variant or add the missing package. This will never work since you use Python 3.8 and them add a PY27 package here...

Fair enough. I have submitted PR for `py-pybonjour`: https://github.com/macports/macports-ports/pull/14781